apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -2137,25 +2152,77 @@ def create_ingestion_dag(
             input_vectors_work_items_per_worker_during_sampling
         )
 
+        task_memory_overhead = {"6Gi": 2, "8Gi": 3, "12Gi": 4, "16Gi": 6, "32Gi": 12}
+
+        GB = 1024 * 1024 * 1024
+
+        # Drop last 2 characters from base and convert string to int
+        def convert_base_to_int(base):
+            return int(base[:-2])
+
+        def ram_requirement_bytes(overhead):
+            return int(task_memory_overhead[overhead]) * this_batch_size_bytes
+
+        kmeans_requirement_bytes = kmeans_batch_size_bytes * 250
+
+        # Function to convert ram_requirement_bytes to ram string (approximates to nearest multiple of 2)
+        def get_scaled_ram_gi(base):
```

Comment:
I think that the logic of `get_scaled_ram_gi, get_scaled_ram_gi_kmeans` is very complicated for what we want to achieve.

We could instead have a util function
```
def scale_resources(min_resource, max_resource, max_input_size, input_size):
    return max(
       min_resource, 
       min(
            max_resource, 
            int(max_resource * input_size / max_input_size)
       )
   )
``` 

And then use it to compute all resources:
```
ingest_cpu = scale_resources(2, 16, DEFAULT_PARTITION_BYTE_SIZE, size_in_bytes)
ingest_ram = scale_resources(2, 16, DEFAULT_PARTITION_BYTE_SIZE, size_in_bytes)
kmeans_cpu = scale_resources(2, 8, DEFAULT_KMEANS_BYTES_PER_SAMPLE, training_sample_size_in_bytes)
ingest_ram = scale_resources(2, 32, DEFAULT_KMEANS_BYTES_PER_SAMPLE, training_sample_size_in_bytes)
```

Reply:
Could you take a look now?